dataland-frontend/src/utils/updateObjectUtils.ts

```diff
@@ -36,8 +37,28 @@ export function modifyObjectKeys(obj: objectType, typeOfModification: string): o
         objectModified[key] = (Math.round(+objectModified[key] * 100 * 100) / 100).toString();
       }
     } else if (typeof objectModified[key] === "object" && key !== "totalAmount") {
-      modifyObjectKeys(objectModified[key] as unknown as objectType, typeOfModification);
+      modifyObjectKeys(objectModified[key] as unknown as ObjectType, typeOfModification);
     }
   }
   return objectModified;
 }
+
+/**
+ * A recursive function that is able to retrieve all values for a provided key in an arbitrarily deeply nested object.
+ * Example: the datasource.reference in the EUTaxonomyDataModel occurs for several data points and you might want to get
+ * a list of all references in one dataset.
+ * @param [obj] object in which it is looking for the value to change
+ * @param [keyToFind] the key which is to be found
+ * @returns all the values corresponding to the key
+ */
+export function findAllValuesForKey(obj: ObjectType, keyToFind: string): Array<string> {
+  return Object.entries(obj).reduce((acc: Array<string>, [key, value]) => {
```

Comment:
Das ist bei mir in der IDE wg. Type-Mismatch rot. To be investigated.

Reply:
Starting to investigate this

Reply:
Maybe just a short comment. In my IDE this error is not shown.

Reply:
Ok forget my last comment.  I have mixed two things up.

I observe the same thing as @stewal19:    It is not an error in my IDE.   IntelliJ states that I am using "Typescript 5.0.4." at the bottom of the IntelliJ window btw.

Reply:
So can this conversation be closed?   I just checked and still it's not red for me.